app/models/server/models/Users.js

```diff
@@ -717,6 +717,15 @@ export class Users extends Base {
 		return this.find(query, options);
 	}
 
+	findByUsernamesIgnoringCase(usernames, options) {
+		const combineUsernames = usernames.join('|^');
```

Comment:
we need to escape user inputted values when using as regex with `escapeRegExp`.

so for this case, it might be easier to do the following:
```javascript
const query = {
	username: {
		$in: usernames.map((username) => new RegExp(escapeRegExp(combineUsernames), 'i')),
	},
};

Reply:
Ah, good catch. Totally agree with that. Thanks!